packages/mapper/lib/mapper.js

```diff
@@ -48,15 +49,33 @@ class Mapper {
 
   setCtx(ctx) {
     this._ctx = ctx;
-    this._ctx.db = new Database(this, 'test');
+    this._ctx.db = this.databases.test;
   }
 
-
   use(_, db) {
-    this._ctx.db = new Database(this, db);
+    if (!(db in this.databases)) {
+      this.databases[db] = new Database(this, db);
+    }
+    this._ctx.db = this.databases[db];
     return `switched to db ${db}`;
   }
 
+  async show(_, arg) {
+    switch (arg) {
+      case 'databases':
+      case 'dbs':
+        const result = await this._serviceProvider.listDatabases('admin');
+        if (!('databases' in result)) {
+          throw new Error('Error: invalid result from listDatabases');
+        }
+        return result.databases.reduce((str, db) => {
+          return `${str}\n${db.name}\t${db.sizeOnDisk}B`;
```

Comment:
Minor thing: `mongo` uses GB instead of B

Reply:
Yeah I noticed that, but I'm not convinced its better. The same `show dbs` command on the old shell just returns 0GB, 0GB, 0GB which i think is less informative than B if they're small. Should we update it to do what `ls` does, i.e. "human readable format" (https://web.archive.org/web/20120507054320/http://codeaid.net/javascript/convert-size-in-bytes-to-human-readable-format-(javascript))

Reply:
I was thinking the same. One downside is the results won't be nicely aligned as they are in `mongo`:

```
> show dbs
100YWeatherSmall   0.108GB
admin              0.000GB
agenda             0.000GB
agendaDb           0.000GB
city               0.002GB
citybike           0.005GB
compass           12.454GB
config             0.000GB
debugging          0.000GB
foo                0.000GB
graphql_guide      0.000GB
gridfs             0.009GB
las_vegas          0.000GB
local              0.000GB
metrics            0.000GB
ships              0.001GB
test               3.084GB
video              0.256GB
```
`mongo` uses decimals, maybe that's good enough?

Reply:
I think they'll be aligned if we use the tab character right? As long as there aren't any exceedingly long db names. We can do some math to make sure the size is rounded to 4-5 digits?